lib/base-compiler.js

```diff
@@ -436,6 +436,14 @@ class BaseCompiler {
         return {compiler: this.compiler, source, options, backendOptions, filters, tools, libraries};
     }
 
+    getCompilationInfo(key, result) {
+        let compilationinfo = Object.assign({}, key, result);
```

Comment:
We can use the spread syntax onthe server code, but I guess this is much clearer in this case :)

Reply:
I think more importantly is to communicate the available things to tool developers, need to make a special mddoc for tools i think

Reply:
Cool. For what it's worth this could be `const` (JS `const` being funny, and only referring to the reference, like `Object& const = {...}` in C++, rather than `const Object &`)